pandas/core/construction.py

```diff
@@ -402,6 +402,24 @@ def extract_array(obj: object, extract_numpy: bool = False) -> Union[Any, ArrayL
     return obj
 
 
+def wrap_if_datetimelike(arr):
```

Comment:
agree with joris that `maybe_*` is pretty much a convention and would go with that nomenclature.

Reply:
maybe_do_x usually means "try to do x and fallback on failure".  i think wrap_if_datetimelike is more precise on this point.

Reply:
ok how about
ensure_datetimelike_index then